### PR TITLE
scan: removed all warnings

### LINKED ISSUE #998: Remove all warnings from scan code
Some warnings are legit. Like non-exhausive pattern matching warnings. Let's clear all the warnings.

### PR DESCRIPTION
**Pull request by tansawit**
Fixes: #998

### CODE REVIEW

**Review by prin-r — COMMENTED** [review]
I'm not sure about renaming parameters by adding _.

File: scan/src/App.re [hunk 1]
```diff
@@ -113,7 +113,7 @@ let make = () => {
          | BlockHomePage => <BlockHomePage />
          | BlockIndexPage(height) => <BlockIndexPage height />
          | ValidatorHomePage => <ValidatorHomePage />
-         | RequestIndexPage(reqID, hashtag) => <RequestIndexPage reqID hashtag />
+         | RequestIndexPage(reqID, hashtag) => <RequestIndexPage reqID _hashtag=hashtag />
```
Comment (sorawit) [inline scan/src/App.re]: If we don't use it, why don't we remove it? Also is the type necessaary?

File: scan/src/components/data-source/DataSourceExecute.re [hunk 2]
```diff
@@ -77,7 +77,7 @@ let parameterInput = (name, placeholder, index, setCalldataArr) => {
       onChange={event => {
         let newVal = ReactEvent.Form.target(event)##value;
         setCalldataArr(prev => {
-          prev->Belt_Array.set(index, newVal);
+          ignore(prev->Belt_Array.set(index, newVal));
```
Comment (sorawit) [inline scan/src/components/data-source/DataSourceExecute.re]: Don't use `Belt_Array.set`.

File: scan/src/components/oracle-script/OracleScriptExecute.re [hunk 3]
```diff
@@ -97,7 +97,7 @@ let parameterInput = (name, placeholder, index, setCalldataArr) => {
       onChange={event => {
         let newVal = ReactEvent.Form.target(event)##value;
         setCalldataArr(prev => {
-          prev->Belt_Array.set(index, newVal);
+          ignore(prev->Belt_Array.set(index, newVal));
```
Comment (sorawit) [inline scan/src/components/oracle-script/OracleScriptExecute.re]: ditto

File: scan/src/reusable/AddressRender.re [hunk 4]
```diff
@@ -1,31 +1,30 @@
 type pos_t =
   | Title
   | Subtitle
-  | Text
-  | None;
+  | Text(bool);
```
Comment (sorawit) [inline scan/src/reusable/AddressRender.re]: What's the semantic of this bool?

File: scan/src/rpc/TxHook.re [hunk 5]
```diff
@@ -320,10 +321,10 @@ module Msg = {
       | Some(value) => "#" ++ (report.requestID |> string_of_int) ++ " " ++ value
       | None => "?"
       }
-    | AddOracleAddress(address) => "ADDORACLEADDRESS DESCRIPTION"
-    | RemoveOracleAddress(address) => "REMOVEORACLEADDRESS DESCRIPTION"
-    | CreateValidator(validator) => "CREATEVALIDATOR DESCRIPTION"
-    | EditValidator(validator) => "EDITVALIDATOR DESCRIPTION"
+    | AddOracleAddress(_address) => "ADDORACLEADDRESS DESCRIPTION"
```
Comment (sorawit) [inline scan/src/rpc/TxHook.re]: Will all be removed?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: scan/src/reusable/AddressRender.re (hunk 4 above)
Comment (sorawit) [reply scan/src/reusable/AddressRender.re]:
```
type pos_text_t =
| A
| B;

type pos_t =
  | Title
  | Subtitle
  | Text(pos_text_t);
```

File: scan/src/reusable/AddressRender.re [hunk 6]
```diff
@@ -1,31 +1,34 @@
-type pos_t =
+type isPointer =
```
Comment (sorawit) [inline scan/src/reusable/AddressRender.re]: use snake_case for type name.

**Review by sorawit — CHANGES_REQUESTED** [review]

**Comment by evilpeach** [comment]
@tansawit Please fix the conflict and change `blockIndexTxsTable` to `BlockIndexTxsTable`.

File: scan/src/components/TxsTable.re [hunk 7]
```diff
@@ -127,15 +127,15 @@ let make = (~txs: array(TxSub.t)) => {
                  </div>
                </div>
              </Col>
-             <Col size=5.>
+             <Col size=3.>
```
Comment (evilpeach) [inline scan/src/components/TxsTable.re]: Don't change flex size

File: scan/src/components/TxsTable.re [hunk 8]
```diff
@@ -127,15 +127,15 @@ let make = (~txs: array(TxSub.t)) => {
...
                {messages
                 ->Belt_List.toArray
-                ->Belt_Array.map(msg =>
-                    <>
+                ->Belt_Array.mapWithIndex((i, msg) =>
+                    <React.Fragment key={(txHash |> Hash.toHex) ++ (i |> string_of_int)}>
                       <VSpacing size=Spacing.sm />
-                      <Msg msg success width=450 />
+                      <Msg msg success width=385 />
```
Comment (evilpeach) [inline scan/src/components/TxsTable.re]: Don't change it too.

**Review by sorawit — DISMISSED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Review by prin-r — APPROVED** [review]

**Pull request merged**